pkg/gatherers/clusterconfig/gather_openstack_dataplane_node_sets.go

```diff
@@ -0,0 +1,111 @@
+package clusterconfig
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/klog/v2"
+
+	"github.com/openshift/insights-operator/pkg/record"
+	"github.com/openshift/insights-operator/pkg/utils/anonymize"
+)
+
+// GatherOpenstackDataplaneNodesets Collects `openstackdataplanenodesets.core.openstack.org`
+// resources from all namespaces
+//
+// ### API Reference
+// None
+//
+// ### Sample data
+// - docs/insights-archive-sample/customresources/dataplane.openstack.org/openstackdataplanenodesets/openstack/openstack-edpm.json
+//
+// ### Location in archive
+// - `customresources/dataplane.openstack.org/openstackdataplanes/{namespace}/{name}.json`
+//
+// ### Config ID
+// `clusterconfig/openstack_dataplane_nodesets`
+//
+// ### Released version
+// - 4.15
+//
+// ### Changes
+// None
+func (g *Gatherer) GatherOpenstackDataplaneNodeSets(ctx context.Context) ([]record.Record, []error) {
+	gatherDynamicClient, err := dynamic.NewForConfig(g.gatherKubeConfig)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	return gatherOpenstackDataplaneNodeSets(ctx, gatherDynamicClient)
+}
+
+func gatherOpenstackDataplaneNodeSets(ctx context.Context, dynamicClient dynamic.Interface) ([]record.Record, []error) {
+	osdpnsList, err := dynamicClient.Resource(osdpnsGroupVersionResource).List(ctx, metav1.ListOptions{})
+	if errors.IsNotFound(err) {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	var records []record.Record
+
+	for i, osdpns := range osdpnsList.Items {
+		records = append(records, record.Record{
+			Name: fmt.Sprintf("customresources/%s/%s/%s/%s",
+				osdpnsGroupVersionResource.Group,
+				osdpnsGroupVersionResource.Resource,
+				osdpns.GetNamespace(),
+				osdpns.GetName(),
+			),
+			Item: record.ResourceMarshaller{Resource: prepareOpenStackDataPlaneNodeSet(&osdpnsList.Items[i])},
+		})
+	}
+
+	return records, nil
+}
+
+func prepareOpenStackDataPlaneNodeSet(data *unstructured.Unstructured) *unstructured.Unstructured {
+	fieldsToRemove := [][]string{
+		{"metadata", "annotations", "kubectl.kubernetes.io/last-applied-configuration"},
+	}
+	fieldsToAnonymize := [][]string{}
+	customFieldsToAnonymize := []string{
+		"ansibleUser", "edpm_sshd_allowed_ranges", "dnsClusterAddresses",
+	}
+	data.Object = removeFields(data.Object, fieldsToRemove)
+	data.Object = anonymizeIpAddresses(data.Object)
+	data.Object = anonymizeFields(data.Object, fieldsToAnonymize)
+	data.Object = anonymizeCustomPathFields(data.Object, customFieldsToAnonymize)
+	data.Object = anonymizeStatusHostNames(data.Object)
```

Comment:
each node object in the node set can have multiple hostnames assocrrated with it too.

there is a hostname subsection 
https://github.com/openstack-k8s-operators/openstack-operator/blob/main/apis/core/v1beta1/openstackcontrolplane_types.go#L361-L370

the AnsibleOpts also contained sensitive info in general but that is a mapping filed with arbiary keys.

mouch of the value comes form processing those value but its harder to Anonymize in general.

at a minium we should be Anonymize ansibleHost

https://github.com/openstack-k8s-operators/openstack-operator/blob/main/apis/core/v1beta1/openstackcontrolplane_types.go#L361-L370

and any fixed ipes that exixits in the networks struct
https://github.com/openstack-k8s-operators/openstack-operator/blob/main/apis/core/v1beta1/openstackcontrolplane_types.go#L361-L370